src/libsyntax/parse/parser/stmt.rs

```diff
@@ -473,4 +492,61 @@ impl<'a> Parser<'a> {
             "this was erroneously allowed and will become a hard error in a future release"
         }).emit();
     }
+
+    fn recover_stmt_local(
+        &mut self,
+        span: Span,
+        attrs: Vec<Attribute>,
+        msg: &str,
+    ) -> PResult<'a, Option<Stmt>> {
+        let local = self.parse_local(attrs.into())?;
+        self.struct_span_err(span, "invalid variable declaration")
+            .span_suggestion_short(
+                span,
+                msg,
+                "let".to_string(),
+                Applicability::MachineApplicable
+            ).emit();
+        return Ok(Some(self.mk_dummy_stmt(span, local)));
+    }
+
+    fn mk_dummy_stmt(&mut self, span: Span, kind: P<Local>) -> Stmt {
+        Stmt {
+            id: DUMMY_NODE_ID,
+            kind: StmtKind::Local(kind),
+            span,
+        }
+    }
+
+    /// Returns `true` if the next token is Brace
+    fn is_next_brace(&self) -> bool {
+        self.look_ahead(1, |t|  t.eq(&token::OpenDelim(token::Brace)))
+    }
+
+    /// Returns `true` if the token is kw::Mut and next one is an Ident
+    fn is_ident_mut(&self) -> bool {
```

Comment:
Tho I appreciate the spirit of writing small functions, this function is small enough that it can be inlined where it is used. :)  (General point that applies to other functions introduced.)